prov/cxi/src/cxip_mr.c

```diff
@@ -363,10 +649,19 @@ static int cxip_mr_enable_opt(struct cxip_mr *mr)
 	if (mr->cntr)
 		le_flags |= C_LE_EVENT_CT_COMM;
 
+	/* RMA events on optimized MRs are not expected (we force standard), but
+	 * if they occur, enable communication events and keep success disabled.
+	 */
+	if (mr->rma_events) {
+		le_flags &= ~C_LE_EVENT_COMM_DISABLE;
+		CXIP_DBG("Optimized MR enabling RMA events (unexpected): mr=%p\n", mr);
+	}
+
 	if (target_relaxed_order) {
 		ib = 1;
-		le_flags |= C_LE_UNRESTRICTED_END_RO |
-			C_LE_UNRESTRICTED_BODY_RO;
+		le_flags |= C_LE_UNRESTRICTED_END_RO | C_LE_UNRESTRICTED_BODY_RO;
+	} else {
+		le_flags |= C_LE_UNRESTRICTED_BODY_RO;
```

Comment:
We alter the pre-existing behavior by adding C_LE_UNRESTRICTED_BODY_RO if target relaxed ordering has not be explicitly requested. This change probably isn't related to the writedata() implementation, and if it not it seems like we should make that else clause a separate PR/fix that can carry reasoning with it.